tests/DataObjectTest.py

```diff
@@ -110,3 +110,28 @@ def __init__(self, a=None, **kw):
     def test_rdfs_comment_property(self):
         a = DataObject(rdfs_comment='Hello')
         self.assertIn('Hello', a.rdfs_comment())
+
+    def test_context_getter(self):
+        a = DataObject(rdfs_comment='Hello')
+        self.assertIsNone(a.context)
+
+    def test_context_setter(self):
+        a = DataObject(rdfs_comment='Hello')
+        a.context = 42
+        self.assertEquals(a.context, 42)
+
+    # TODO: only test when raise here
+    def test_datatype_property(self):
+        print(DataObject(rdfs_comment='Hello').DatatypeProperty(dict()))
```

Comment:
What's the goal of passing a dict() to DatatypeProperty?

Reply:
Here is only for enter in a special case of a sub method. Do you think I should populated with some others data for this case ?

Reply:
Catching TypeError is a bit of a hack...Python throws a TypeError when there are missing required arguments to a method and then we create a _partial_property which DataObject.__setattr__ picks up on and actually creates the property. That all is because I wanted to keep the existing behavior of DatatypeProperty, etc. where you could declare a property without an assignment and Python doesn't have built-in function currying.

That said, I think we can drop the behavior where you declare a property in the `__init__`. As long as you can still do `SomeClass.DatatypeProperty`